modules/OFConnectionManager2/module/src/cxn_instance.c

```diff
@@ -825,6 +825,164 @@ ind_cxn_unblock_barrier(connection_t *cxn)
     }
 }
 
+/**
+ * increase the async pending cnt
+ * Set the bundle async op pending flag
+ * @param cxn
+ */
+void
+ind_cxn_block_async_op(connection_t *cxn)
+{
+    cxn->async_pending_cnt++;
+}
+
+void
+indigo_cxn_block_async_op(indigo_cxn_id_t cxn_id)
+{
+    connection_t *cxn = ind_cxn_id_to_connection(cxn_id);
+    if (cxn != NULL) {
+        ind_cxn_block_async_op(cxn);
+    }
+}
+
+/**
+ * Clear the async op pending flag
+ * @param cxn
+ */
+void
+ind_cxn_unblock_async_op(connection_t *cxn)
+{
+    if (cxn->async_pending_cnt >  0) {
+        cxn->async_pending_cnt--;
+    }
+}
+
+void
+indigo_cxn_unblock_async_op(indigo_cxn_id_t cxn_id)
+{
+    connection_t *cxn = ind_cxn_id_to_connection(cxn_id);
+    if (cxn != NULL) {
+        ind_cxn_unblock_async_op(cxn);
+    }
+}
+
+/**
+ * Set the iter async op pending flag
+ * @param cxn
+ */
+void
+ind_cxn_block_iter_async_op(connection_t *cxn)
+{
+    cxn->iter_async_pendingf = true;
+}
+
+void
+indigo_cxn_block_iter_async_op(indigo_cxn_id_t cxn_id)
+{
+    connection_t *cxn = ind_cxn_id_to_connection(cxn_id);
+    if (cxn != NULL) {
+        ind_cxn_block_iter_async_op(cxn);
+    }
+}
+
+/**
+ * Clear the iteration async op pending flag
+ * @param cxn
+ */
+void
+ind_cxn_unblock_iter_async_op(connection_t *cxn)
+{
+    cxn->iter_async_pendingf = false;
+}
+
+void
+indigo_cxn_unblock_iter_async_op(indigo_cxn_id_t cxn_id)
+{
+    connection_t *cxn = ind_cxn_id_to_connection(cxn_id);
+    if (cxn != NULL) {
+        ind_cxn_unblock_iter_async_op(cxn);
+    }
+}
+
+/**
+ * Check whether connection is in the iter task pending status
+ * @param cxn_id connection id
+ * @param iter_concurrent allow concurrent iteration without blocking
+ */
+bool
+ind_cxn_iter_task_should_yield(connection_t *cxn)
```

Comment:
i'm missing something here also; why are we getting into the iter task?
also, why do we have "may_yield" and "should_yield"?  there should only be one such yield API...  but going one level up, i'm thinking we are only handling the bundle iteration, so why even have another set of APIs for the iter task?